### PR TITLE
Added Lens Logic, and fixed Ice Cavern

### PR DESCRIPTION
**Pull request by Squidkingdom**
Ice Cavern needed bomb/scale and another bottle in addition to ruto's letter.

Lens logic can be added by calling either `lens('all')` or 'lens('semi') to describe Everywhere lens and Wasteland lens logic respectivily.

Still show merge conflicts for everything, but i ran it through a diff checker, nothing has gone backwards, just my new feature. I think my IDE is doing something janky

### CODE REVIEW

**Comment by dombarrieau** [comment]
Any reason the XML files were added? I'm not familiar with it. Also you have at least one large portion of commented code. If it's for a future implementation, I'd rather it just not be included in this commit

**Comment by Squidkingdom** [comment]
Dont know what your talking about as for commented code, XML files are cause my Gitignore screwed up.

**Comment by Squidkingdom** [comment]
Should be good now, ignore the messiness XD

**Comment by AlternativeSquid** [comment]
Oh ok yeah no that's useless, left over from when I was experimenting with putting that function in main and not chests. Ill fix it when I get home.

**Comment by Squidkingdom** [comment]
Oh ok yeah no that's useless, completely forgot about that, think it's left over from when I was experimenting with putting that function in main and not chests. Ill fix it when I get home.

File: script/main.js [hunk 1]
```diff
@@ -864,7 +872,39 @@ function populateItemconfig() {
         row.appendChild(rowitem);
     }		
 }
-
+function returnLens(){
+    return items.lens
+}
+// function lens(loc){
```
[the remark this answers is not in the document]
Comment (Squidkingdom) [reply script/main.js]: This function has been removed

**Comment by Squidkingdom** [comment]
Removed the useless function! Anything else you see that needs changed?

**Comment by dombarrieau** [comment]
After looking into the code more, there were some changes I did not agree with (adding an extra tab to all the dungeon return lines). It also created some logic errors (spirit Map Chest does not need magic and dins if accessible as child, longshot can be used in Dodongo's to get to the Bomb Bag chest, Adult Fishing requires beans or hookshot to get to). It also had a few conflicts with TR's recent changes. It also would have caused the commit history to be quite messy, which would then required them to be squashed. I put all your good changes into a single commit and got it merged, so I will close this one. In the future, try to make your changes smaller. Try not to fix a whole lot of things in the same PR/commits

**Pull request closed without merging**